python/tskit/trees.py

```diff
@@ -4024,10 +4206,127 @@ def trim(self, record_provenance=True):
         tables.trim(record_provenance)
         return tables.tree_sequence()
 
-    def draw_svg(self, path=None, **kwargs):
-        # TODO document this method, including semantic details of the
-        # returned SVG object.
-        draw = drawing.SvgTreeSequence(self, **kwargs)
+    def draw_svg(
+        self,
+        path=None,
+        *,
+        size=None,
+        tree_height_scale=None,
+        max_tree_height=None,
+        node_labels=None,
+        mutation_labels=None,
+        node_attrs=None,
+        mutation_attrs=None,
+        edge_attrs=None,
+        node_label_attrs=None,
+        mutation_label_attrs=None,
+        root_svg_attributes=None,
+        style=None,
+    ):
+        """
+        Return an SVG representation of a tree sequence.
+
+        When working in a Jupyter notebook, use the ``IPython.display.SVG`` function
+        to display the SVG output from this function inline in the notebook::
+
+            >>> SVG(tree.draw_svg())
+
+
+        .. note::
+
+            The visual elements in the svg are
+            `grouped <https://www.w3.org/TR/SVG2/struct.html#Groups>`_
+            for easy styling and manipulation. The entire visualization with trees and X
+            axis is contained within a group of class ``tree-sequence``. Each tree in
+            the displayed tree sequence is contained in a group of class ``tree``, as
+            described in :meth:`Tree.draw_svg`, so that visual elements pertaining to one
+            or more trees targetted as documented in that method. For instance, the
+            following style will change the colour of all the edges of the *initial*
+            tree in the sequence and hide the internal node labels in *all* the trees
+
+            .. code-block:: css
+
+                .tree.t0 .edges {stroke: blue}
+                .tree .labels .nodes text:not(.sample) {visibility: hidden}
+
+            See :meth:`Tree.draw_svg` for further details.
+
+        :param str path: The path to the file to write the output. If None, do not write
+            to file.
+        :param size: A tuple of (width, height) giving the width and height of the
+            produced SVG drawing in abstract user units (usually interpreted as pixels on
+            display). If None, defaults to a width of
+            :math:`200 \\times \\textrm{num_trees}` and a height of :math:`200`.
+        :type size: tuple(int, int)
+        :param str tree_height_scale: Control how height values for nodes are computed.
+            If this is equal to ``"time"``, node heights are proportional to their time
+            values (this is the default). If this is equal to ``"log_time"``, node
+            heights are proportional to their log(time) values. If it is equal to
+            ``"rank"``, node heights are spaced equally according to their ranked times.
+        :param str,float max_tree_height: The maximum tree height value for each tree in
```

Comment:
Why do we want to add this parameter to TreeSequence.draw? I don't think it makes any sense to have different Y-scales per tree. We want to have a Y-axis showing time at some point, and having this option would break that.

Reply:
It was in before, but happy to remove. The only reason is you might want to compare the topology between trees in a ts without caring about the heights, and so scale all trees to the same height so that the short ones are still readable. So height=rank might make sense here.

Reply:
Let's remove it for now to simplify things here.

Reply:
Removed from the `svg_draw()` exposed function, although I left it in the underlying one for simplicity.